pyro/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,123 @@
+        therefore enabled by default.
+    :raises: ValueError if the shape of mean or sigma is not supported.
+    """
+
+    reparameterized = True
+
+    def __init__(self, mu, sigma, batch_size=None, is_cholesky=False, use_inverse_for_batch_log=False, *args, **kwargs):
+        self.mu = mu
+        self.output_shape = mu.size()
+        self.use_inverse_for_batch_log = use_inverse_for_batch_log
+        self.batch_size = batch_size if batch_size is not None else 1
+        if not is_cholesky:
+            self.sigma = sigma
+            self.sigma_cholesky = Variable(torch.potrf(sigma.data)) if torch.__version__ < '0.3.0' else torch.potrf(
+                sigma)
+        else:
+            self.sigma = torch.mm(sigma.transpose(0, 1), sigma)
+            self.sigma_cholesky = sigma
+        if mu.dim() > 1:
+            raise ValueError("The mean must be a vector, but got mu.size() = {}".format(mu.size()))
+        if not sigma.dim() == 2:
+            raise ValueError("The covariance matrix must be a matrix, but got sigma.size() = {}".format(mu.size()))
+
+        super(MultivariateNormal, self).__init__(*args, **kwargs)
+
+    def batch_shape(self, x=None):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.batch_shape`
+        """
+        mu = self.mu.expand(self.batch_size, *self.mu.size()).squeeze(0)
+        if x is not None:
+            if x.size()[-1] != mu.size()[-1]:
+                raise ValueError("The event size for the data and distribution parameters must match.\n"
+                                 "Expected x.size()[-1] == self.mu.size()[0], but got {} vs {}".format(
+                                    x.size(-1), mu.size(-1)))
+            try:
+                mu = mu.expand_as(x)
+            except RuntimeError as e:
+                raise ValueError("Parameter `mu` with shape {} is not broadcastable to "
+                                 "the data shape {}. \nError: {}".format(mu.size(), x.size(), str(e)))
+
+        return mu.size()[:-1]
+
+    def event_shape(self):
+        """
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.event_shape`
+        """
+        return self.mu.size()[-1:]
+
+    def sample(self):
+        """
+        Generate a sample with the specified covariance matrix and mean.
+        Differentiation wrt. to Sigma is only supported on PyTorch version 0.3.0 or higher.
+        Ref: :py:meth:`pyro.distributions.distribution.Distribution.sample`
+
+        """
+        batch_size = self.batch_size
+        uncorrelated_standard_sample = Variable(torch.randn(batch_size, *self.mu.size()).type_as(self.mu.data))
+        transformed_sample = self.mu + torch.mm(uncorrelated_standard_sample, self.sigma_cholesky)
+        return transformed_sample if self.reparameterized else transformed_sample.detach()
+
+    def batch_log_pdf(self, x, normalized=True):
```

Comment:
This `normalized` flag is nice to have. Right now we don't have the plumbing to support it as an argument to `batch_log_pdf()`, but we can support it as an argument to the `__init__` method. Could you move it there and make an instance attribute `.normalized`?